source/file_test.go

```diff
@@ -0,0 +1,348 @@
+package source_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/mercari/grpc-federation/source"
+)
+
+func TestFile_FindLocationByPos(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		desc     string
+		pos      source.Position
+		expected *source.Location
+	}{
+		{
+			desc: "find nothing",
```

Comment:
Currently, `testdata/service.proto` is read directly from each tests. So it may be possible to pass the file name and read that file. In this case, the `testdata` prefix should be omitted and `service.proto` or `service` would be fine.